### DISCUSSION TITLE
Is accelerator boards support on the roadmap?

### DISCUSSION THREAD

**Discussion by jman-schief** [BlitterStudio/amiberry#1365] [Q&A] [answered]
Hi, I was wondering if in the roadmap of Amiberry there might be support for more expansions such as 030 (and above) accelerator boards (I understand that PPC support is currently [not feasable](https://github.com/BlitterStudio/amiberry/issues/527)).

thanks

**Comment by solskogen** [comment]
030 and 040 is already available on master, while 060 is supported on preview.

**Reply by jman-schief** [reply]
I'm using the latest preview and can set the main CPU to 030/040/060 but can't figure out how to enable an accelerator board. To be clear I mean expansion boards such as classic Blizzards or other 030/040/060 accellerator boards.

Am I missing anything?

**Reply by solskogen** [reply]
Correct, the different accelerator cards isn't supported. What's the use case for it, in a emulator I mean when you already have support for CPU's up to 060?

**Reply by jman-schief** [reply]
> What's the use case for it, in a emulator I mean when you already have support for CPU's up to 060?

That's an interesting question but probably orthogonal to the question asked here. I might reply with a counter-question: is having a 040 as main CPU _exactly_ the same as having it on a accelerator board? Since WinUAE (which Amiberry is based on) supports them, I _think_ there are compelling reasons for doing that, apart obvious preservation purposes.

**Reply by solskogen** [reply]
I just don't see what they provide which is not already there (expect for PPC that is)

**Reply by giantclambake** [reply]
The only case wherein it may matter, ...at least afaict..,  is if some Amiga software specifically needed the accelerator ID, rom file contents/functions, or something along those lines for the software to work. Correct me if I'm wrong, but the only Amiga software I can think of like that, is the setup/utility software that came with those hardware accelerators ... which we don't need/use, because we have analogous emulator options/settings to meet those ends ;)

The PPC case is a good example ~ I'm pretty sure there's Amiga software out there that require this CPU option (starts with the actual kickstart ROM makeup iirc), and in real hardware you'd need a PPC board or PPC accelerator to meet those ends -- in amiberry, all we need provide is an emulation of the PPC  CPU, as that's all this software is looking for ...ie; it's not looking for any specific PPC accelerator....just the CPU functionality. Same goes for other Amiga software... it's not searching for any specific  030/040/060 accelerator card, just the (emulated) Amiga capabilities and CPU type.

IRL... (fancy that ;)... this is the 3rd or 4th time this year, I've seen some post on EAB, posturing that amiberry is 'limited' and/or 'incomplete' compared to fs-uae/WinUAE, because it doesn't contain emulated accelerator options (or shaders, etc). This is why jman-schief is here ~ I posited on EAB that one possible reason the accelerator options are not present in amiberry, is that no user ever asked about same, (as we don't really need it), and/or no user ever proffered the case for a specific need for a particular accelerator card, that some Amiga software needed to run correctly ; now the question's been asked, but the case need has yet to be demonstrated/proven.

Personally speaking (and why I've asked midwan the question below), is that the greyed out 'Accelerator Board Settings' in the Expansion panel, I consider to be an 'eye sore' wrt the amiberry GUI...ie; it's just there, it doesn't do anything, in normal practice it isn't needed, and even if that section were functional ...like solskogen points out... what benefits or gains would it provide, that we don't already have? I can't determine any need .... except of course, to make amiberry looks as replete as fs-uae/WinUAE are ... it is what it is ;)

That section is greyed out, under the auspices of "_Please note that many of the options here are not yet fully functional, since they are currently stripped down in Amiberry. They may be added in future versions, but for now only the most commonly used options are available._" .... query: When would accelerator board settings, become a "most commonly used option"?  I really can't see it... and because I'm a cut-throat 500 player (card game ;), I'm wondering if the Accelerator Board Settings section should removed altogether ...ie; "Please note that many of the options here are not present in Amiberry, as they are not required and Amiberry provides other means to control the related option one is used to seeing here in other emulator"...

**Comment by giantclambake** [comment]
@solskogen  ~ question relates to accelerator board emulation, not CPU type ;) AFAIK such hasn't been implemented yet in amiberry (either branch).

@jman-schief 

Ticket #527 is old, and doesn't really reflect the current state of play.  Since that time, with the addition of amiberry v6.x (preview branch), it should be possible to provide PPC emulation support in the future, for more powerful host systems (apple CPU and x86-64). However, as midwan notes in that ticket, it would negatively impact the performance of the emulation on lower powered Arm based hardware, so this may never be part of amiberry v5.x, unless more powerful versions of these boards become available.

@midwan ~ what are the plans, if any, to provide accelerator board emulation? Likewise for PPC emulation in `preview` ~ is that on the radar?

**Comment by midwan** [comment]
Support for accelerator boards could come in future versions, but it's not a high priority at the moment. The way I see it, it's significant work with very little actual benefit (considering the functionality added vs the complexity involved). PPC support could happen for the faster boards, it's still not feasible for the Raspberry Pi for example.

**Comment by giantclambake** [comment]
> "considering the functionality added vs the complexity involved"

You've piqued my curiosity --- precisely what functionality is added?  (I had a read of cpuboards.c and saw the complexity =)

**Comment by midwan** [comment, marked as answer]
Just to update this:
The next release of Amiberry includes support for accelerator boards, as well as the various expansion options available on WinUAE (custom gfx cards, etc).
